openresty-patches/patches/1.21.4.1/nginx-1.21.4_03-stream_upstream_client_certificate_and_ssl_verify.patch

```diff
@@ -0,0 +1,68 @@
+diff --git a/nginx-1.21.4/src/stream/ngx_stream_proxy_module.c b/nginx-1.21.4/src/stream/ngx_stream_proxy_module.c
+index b11c288..bba27f0 100644
+--- a/nginx-1.21.4/src/stream/ngx_stream_proxy_module.c
++++ b/nginx-1.21.4/src/stream/ngx_stream_proxy_module.c
+@@ -8,6 +8,9 @@
+ #include <ngx_config.h>
+ #include <ngx_core.h>
+ #include <ngx_stream.h>
++#if (NGX_STREAM_LUA_KONG)
++#include <ngx_stream_lua_kong_module.h>
++#endif
+ 
+ 
+ typedef struct {
+@@ -821,8 +824,18 @@ ngx_stream_proxy_init_upstream(ngx_stream_session_t *s)
+ 
+ #if (NGX_STREAM_SSL)
+ 
++#if (NGX_STREAM_LUA_KONG)
++
++    if (pc->type == SOCK_STREAM && pscf->ssl
++        && !ngx_stream_lua_kong_get_proxy_ssl_disable(s))
++    {
++
++#else
++
+     if (pc->type == SOCK_STREAM && pscf->ssl) {
+ 
++#endif
++
+         if (u->proxy_protocol) {
+             if (ngx_stream_proxy_send_proxy_protocol(s) != NGX_OK) {
+                 return;
+@@ -1085,7 +1098,11 @@ ngx_stream_proxy_ssl_init_connection(ngx_stream_session_t *s)
+         return;
+     }
+ 
++#if (NGX_STREAM_LUA_KONG)
++    if (pscf->ssl_server_name || ngx_stream_lua_kong_get_upstream_ssl_verify(s, pscf->ssl_verify)) {    
++#else
```

Comment:
add a blank line before `#else`?

Reply:
why this code style?

Reply:
I am not sure about it, but some nginx code use this style.